pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -196,6 +201,15 @@ def restore_loops(self) -> None:
         self.trainer.fit_loop.global_step = self._loaded_checkpoint["global_step"]
         self.trainer.fit_loop.current_epoch = self._loaded_checkpoint["epoch"]
 
+        assert self.trainer.state.fn is not None
+        state_dict = self._loaded_checkpoint.get("loops")
+        if state_dict is not None and self.trainer.state.fn != TrainerFn.TUNING:
```

Comment:
Why this change? This has the drawback of losing traceability.

I don't expect anybody subclassing the trainer and implementing an entirely new base loop from scratch.

Reply:
but this just ensures that we load the state_dict only for the loop that is required based on TrainerFn. else whenever we call validate/test/fit... it will reload all the loop states every time.

Reply:
Sorry, I'm referring to using `getattr` over `self.trainer.fit_loop...`

Reply:
just to avoid all those repetitive if-else statements.